### PR TITLE
Add torch_nntile.nn.SDPA and attention weight layout converters

### PR DESCRIPTION
**Pull request by Muxas**
## Summary

Adds `torch_nntile.nn.SDPA` / `sdpa_eager` for FP32 scaled dot-product attention on `device="nntile"`, matching the NNTile `sdpa_eager` decomposition. Also adds `torch_nntile.nn.weight_layout` helpers for HF ↔ NNTile attention weight conversion (for future GPT-2 integration).

**Runtime modes:** SDPA works in both **eager** and **graph** modes.

## Bugbot fixes

1. **Mask axis order:** Mask shape is now `[q_seq, k_seq]` (dim0 = query, dim1 = key), matching `tensor::mask_scalar` and GPT-2 `attn_mask` / `sdpa_causal_mask_bool_fill`.
2. **`grad_out` validation:** `sdpa_backward` rejects `grad_out` tensors whose shape does not match `Q`.

## API

- `torch_nntile.nn.SDPA(batch_ndim=2)` — `nn.Module` wrapper
- `torch_nntile.nn.sdpa_eager(q, k, v, mask=None, batch_ndim=2)` — functional
- `torch_nntile.nn.weight_layout` — weight converters

## v1 limits

- FP32 only; Q/K/V on `device="nntile"` in layout `[batch..., seq, head_size]`
- Optional BOOL mask `[q_seq, k_seq]` on CPU or nntile
- Fixed scale `1/sqrt(head_size)`
- No dropout, causal flag, GQA, or custom scale

## Tests

- `torch_nntile/tests/test_sdpa_parity.py` — parity, graph mode, mask axis order, `grad_out` shape check
- `torch_nntile/tests/test_attn_weight_layout.py` — weight round-trips

<div><a href="https://cursor.com/agents/bc-b0131342-cfb1-4577-8fd3-0b0dcbb45d74"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-b0131342-cfb1-4577-8fd3-0b0dcbb45d74"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by Muxas** [comment]
@cursor review

**Comment by Muxas** [comment]
@cursor review
